pkg/apis/gcp/validation/shoot.go

```diff
@@ -84,22 +85,30 @@ func validateVolume(vol *core.Volume, fldPath *field.Path) field.ErrorList {
 func ValidateWorkersUpdate(oldWorkers, newWorkers []core.Worker, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for i, newWorker := range newWorkers {
-
 		workerFldPath := fldPath.Index(i)
-		for _, oldWorker := range oldWorkers {
-			if newWorker.Name == oldWorker.Name {
-				if validation.ShouldEnforceImmutability(newWorker.Zones, oldWorker.Zones) {
-					allErrs = append(allErrs, apivalidation.ValidateImmutableField(newWorker.Zones, oldWorker.Zones, workerFldPath.Child("zones"))...)
-				}
-				break
-			}
+		oldWorker, found := getWorkerByName(newWorker.Name, oldWorkers)
```

Comment:
There is already a similar helper func in gardener/gardener - https://github.com/gardener/gardener/blob/f8f5ed365096290d5437b80d860b3d602499e9c8/pkg/apis/core/helper/helpers.go#L248-L256.

Would it be possible to reuse the existing helper func?

```suggestion
		oldWorker := helper.FindWorkerByName(oldWorkers, newWorker.Name)
```

Reply:
Done.

/merge squash